source/slang/lower-to-ir.cpp

```diff
@@ -3446,6 +3446,8 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
     {
         if (auto innerFuncDecl = genDecl->inner->As<FuncDecl>())
             return lowerFuncDecl(innerFuncDecl);
+        else if (auto innerStructDecl = genDecl->inner->As<StructDecl>())
```

Comment:
I'm wondering if this should just call `dispatch` on the inner declaration, so that it can handle all cases (e.g., we also have generic `interface`s and `extension`s to deal with). I don't know the consequences of such a change, though.